src/components/picker/index.js

```diff
@@ -366,98 +393,134 @@ class Picker extends Component {
+    const {value} = this.state;
 
     if (useNativePicker) {
       return <NativePicker {...this.props}/>;
     }
 
-    if (_.isFunction(renderPicker)) {
-      const {value} = this.state;
-
-      return (
-        <View left>
-          <Button {...customPickerProps} link onPress={this.handlePickerOnPress} testID={testID}>
-            {renderPicker(value, this.getLabel(value))}
-          </Button>
-          {this.renderExpandableModal()}
-        </View>
-      );
-    }
+    // if (_.isFunction(renderPicker)) {
+    //   const {value} = this.state;
+
+    //   return (
+    //     <PickerContext.Provider value={this.getContextValue()}>
+    //       <View left>
+    //         <Button {...customPickerProps} link onPress={this.handlePickerOnPress} testID={testID}>
+    //           {renderPicker(value, this.getLabel(value))}
+    //         </Button>
+    //         {this.renderExpandableModal()}
+    //       </View>
+    //     </PickerContext.Provider>
+    //   );
+    // }
 
     const textInputProps = TextField.extractOwnProps(this.props);
     const label = this.getLabelValueText();
     const {paddings, margins, positionStyle} = modifiers;
 
+    const modalProps = {
+      animationType: 'slide',
+      transparent: Constants.isIOS && this.props.enableModalBlur,
+      enableModalBlur: Constants.isIOS && this.props.enableModalBlur,
+      onRequestClose: this.props.topBarProps?.onCancel,
+      onShow,
+      ...this.props.pickerModalProps
+    };
+
     return (
-      <TextField
-        ref={forwardedRef}
-        {...textInputProps}
-        containerStyle={[paddings, margins, positionStyle, containerStyle]}
-        {...this.getAccessibilityInfo()}
-        importantForAccessibility={'no-hide-descendants'}
-        value={label}
-        expandable
-        renderExpandable={this.renderExpandableModal}
-        onToggleExpandableModal={this.toggleExpandableModal}
-        selection={Constants.isAndroid ? {start: 0} : undefined}
-      />
+      <PickerContext.Provider value={this.getContextValue()}>
+        <ExpandableOverlay
+          ref={this.pickerExpandable}
+          modalProps={modalProps}
+          expandableContent={this.renderExpandableModal()}
+          renderCustomOverlay={renderCustomModal ? this.renderCustomModal : undefined}
+          testID={renderCustomModal ? testID : undefined}
+          {...customPickerProps}
+        >
+          {renderPicker ? (
+            renderPicker(value, this.getLabel(value))
+          ) : (
+            <TextField
+              ref={forwardedRef}
+              {...textInputProps}
+              containerStyle={[paddings, margins, positionStyle, containerStyle]}
+              {...this.getAccessibilityInfo()}
+              importantForAccessibility={'no-hide-descendants'}
+              value={label}
+              // expandable
+              // renderExpandable={this.renderExpandableModal}
+              // onToggleExpandableModal={this.toggleExpandableModal}
```

Comment:
Do we want to forbid these (`false`, `noop`) so the user cannot do it within `textInputProps`?